### PR TITLE
tests: automate the release integration-test checklist as end-to-end tests driving

### PR DESCRIPTION
**Pull request by AlekseyManetov**
### Proposed Changes

This PR introduces automation for the [integration tests checklist](https://github.com/camunda/camunda-modeler/blob/develop/docs/.project/INTEGRATION_TEST.md) using Playwright E2E tests.
Most of the checklist is covered, with the exception of these gaps:

## Not covered vs. the checklist

**Needs a running engine / RPA worker:**
- BPMN + Forms **deployment** (deploy bpmn + form).
- **Engine connection** (Local c8run / SaaS / Self-Managed): deploy, start an instance, Operate link, complete the user task in Tasklist.
- RPA **worker "connected"** state and worker-config-persists-after-restart.
- RPA **test execution** (run the script, output in the panel).

**Out of Playwright's reach — stays manual:**
- **Installers** (DMG / archive: download, install, launch).
- **File-type bindings** registration (Windows / Linux / macOS).
- **Double-click** a file in the OS to open it; **drag-and-drop** a file onto the app.
- **OS-native menus** displayed, **restore workspace** on reopen.
- **Accessibility** (axe / "Accessibility violations" modal) — removed; needed a CSP-bypass hack.
- OS delivering the literal keystroke for shortcuts (we drive the bound command, not the OS keypress).
- *"Exports open in browser"* — replaced by golden-baseline comparison (stronger, automatic).

**Automatable but deliberately deferred (small gaps):**
- Exact reference **`test.bpmn`** — build-from-scratch models a representative diagram, not the pixel-exact reference. Intentionally simplified for e2e tests. Currently tested diagram:
image

- **RPA cut/copy/paste** — Monaco's clipboard items are Electron native-roles, not driveable via the shortcut seam.
- **RPA "click test-tube → test dialog opens"** — the RunDialog needs worker context to mount (button visibility is covered).

### Checklist

Ensure you provide everything we need to review your contribution:

* [ ] Contribution __meets our [definition of done](https://github.com/bpmn-io/.github/blob/main/resources/DEFINITION_OF_DONE.md)__
* [ ] Pull request __establishes context__
  * [ ] __Link to related issue(s)__, i.e. `Closes {LINK_TO_ISSUE}` or `Related to {LINK_TO_ISSUE}`
  * [ ] __Brief textual description__ of the changes
  * [ ] __Screenshots or short videos__ showing UI/UX changes
  * [ ] __Steps to try out__, i.e. [using the `@bpmn-io/sr` tool](https://github.com/bpmn-io/sr)

### CODE REVIEW

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -52,8 +62,54 @@ jobs:
...
       if: ${{ runner.OS == 'Linux' }}
-      uses: codecov/codecov-action@v7
+      uses: codecov/codecov-action@v6
       with:
         fail_ci_if_error: true
       env:
         CODECOV_TOKEN: ${{ steps.secrets.outputs.CODECOV_TOKEN }}
+
+
+  e2e-tests:
```
Comment (barmac) [inline .github/workflows/CI.yml:62]: @copilot Placing the e2e tests after the CI means that the tests are sequential. I'd rather have a separate actions for this even if we need to bundle preload and client twice. Let's parallelize if possible.

**Review by barmac — CHANGES_REQUESTED** [review]
It looks that even a test failure is marked as success, cf. https://github.com/camunda/camunda-modeler/actions/runs/27834648029/job/82379629755?pr=6000

Perhaps @copilot could fix that.

File: test/e2e/harness/electron-app.js [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+   */
+  async close() {
+    await this.electronApp.evaluate(({ dialog }) => {
+      dialog.showMessageBox = async (...args) => {
+        const options = args.find(arg => arg && Array.isArray(arg.buttons)) || {};
+        const index = (options.buttons || []).findIndex(label => /don'?t save/i.test(String(label)));
+
+        return { response: index === -1 ? 0 : index };
+      };
+    }).catch(() => {});
```
[the remark this answers is not in the document]
Comment (AlekseyManetov) [reply test/e2e/harness/electron-app.js:215]: hypothetical assumption

**Comment by AlekseyManetov** [comment]
> Fixed in the latest commit. The root cause was `retries: CI ? 2 : 0` in `playwright.config.js` — when a test failed but passed on a later retry, Playwright marked it "flaky" and exited with code 0, so the CI job showed success. Removed retries entirely (`retries: 0`) so any test failure now fails the CI job immediately.

I've reverted the Copilot change since we were occasionally hitting flaky test behavior caused by complex interactions on the canvas (e.g., rendering race conditions, losing focus after connection checks, etc.). Sometimes it's incredibly tricky to bulletproof every edge case in UI tests, so adding retries helps us bypass these transient issues by re-running the failed specs. I ran the CI pipeline 6 times with retries: 2 and didn't get a single failure. Looks like this definitely does the trick to keep our runs stable.

**Comment by jarekdanielak** [comment]
I ran the tests locally a couple of times. I got "should report a lint violation and clear it when fixed" fail once, and this guy failing quite consistently:

```
should validate components against the selected platform version

Error: expect(locator).toContainText(expected) failed

Locator: locator('.linting-tab-item').filter({ hasText: 'Expression' }).first()
Expected substring: "not supported by Camunda Platform 8.0"
Timeout: 5000ms
Error: element(s) not found

Call log:
  - Expect "toContainText" with timeout 5000ms
  - waiting for locator('.linting-tab-item').filter({ hasText: 'Expression' }).first()


  276 |     await app.step('the component is flagged as unsupported at 8.0', async () => {
  277 |       await expect(problems.items().filter({ hasText: 'Expression' }).first())
> 278 |         .toContainText('not supported by Camunda Platform 8.0');
      |          ^
  279 |     });
```

**Comment by jarekdanielak** [comment]
Great job, @AlekseyManetov! 🎖️

If we can this stable, this is a great piece of automation.

---

Seems the setup part is the most time costly. I've been looking for things that can be performed in the same app launch, and found one so far: https://github.com/camunda/camunda-modeler/pull/6000/changes/f0884293644dc2967056804d60a4977fef1b03e4

Hope I'm not messing things up 😄

**Comment by jarekdanielak** [comment]
"should validate components against the selected platform version" failed, because I had c8run running locally. Since we don't test stuff against a cluster, we can run tests in offline mode, cf. https://github.com/camunda/camunda-modeler/pull/6000/commits/1d2c93a89a895f5622fa9c007c446f6ad63816b2

**Comment by AlekseyManetov** [comment]
> Seems the setup part is the most time costly. I've been looking for things that can be performed in the same app launch, and found one so far: https://github.com/camunda/camunda-modeler/commit/f0884293644dc2967056804d60a4977fef1b03e4

> Hope I'm not messing things up 😄

Yes, it's a good observation and improvement. There is no point in setting up a new environment for each type of export, so it makes sense here.

> "should validate components against the selected platform version" failed, because I had c8run running locally. Since we don't test stuff against a cluster, we can run tests in offline mode, cf. https://github.com/camunda/camunda-modeler/commit/1d2c93a89a895f5622fa9c007c446f6ad63816b2

It could also make the tests more stable. Previously, I observed behavior where a completed lint check and report would cause the diagram to blur — which makes some tests flaky.

Thanks for these additions!

**Comment by barmac** [comment]
I've rebased this on top of `main`.

**Review by barmac — APPROVED** [review]
Let's merge it and see what happens.
